adapter/internal/discovery/xds/server.go

```diff
@@ -716,6 +731,12 @@ func cleanMapResources(apiIdentifier string, organizationID string, toBeDelEnvs
 	delete(orgIDOpenAPIEndpointsMap[organizationID], apiIdentifier)
 	delete(orgIDOpenAPIEnforcerApisMap[organizationID], apiIdentifier)
 
+	vHost, err := ExtractVhostFromAPIIdentifier(apiIdentifier)
+	if err != nil {
+		logger.LoggerXds.Error("Error extracting vhost from apiIdentifier. Continue cleaning maps: ", err)
```

Comment:
Can we proceed without a vhost or should we add a return here ?

Reply:
The rate limiter map is moved to the else statement and continues deleting other maps.